mithril.d.ts

```diff
@@ -596,7 +731,32 @@ declare module _mithril {
 		*
 		* @see m.component
 		*/
-		view(ctrl: T): MithrilVirtualElement<T>;
+		view(ctrl: T): MithrilVirtualElement;
+	}
+
+	/**
+	* This represents a parameterized MithrilComponent with up to six parameters,
+	* intended to be instantiated via m.component().
+	*
+	* @see m
+	* @see m.component
+	*/
+	interface ParameterizedMithrilComponent<TController extends MithrilController, T1, T2, T3, T4> {
```

Comment:
I'm okay with this, as long as there's a base case of `...args: any[]` to soak up anything else. Pull this out into its own separate PR.